htdocs/core/lib/functions.lib.php

```diff
@@ -8723,6 +8723,9 @@ function dol_htmlentitiesbr($stringtoencode, $nl2brmode = 0, $pagecodefrom = 'UT
 		$newstring = strtr($newstring, array('&' => '__PROTECTand__', '<' => '__PROTECTlt__', '>' => '__PROTECTgt__', '"' => '__PROTECTdquot__'));
 		$newstring = dol_htmlentities($newstring, ENT_COMPAT, $pagecodefrom); // Make entity encoding
 		$newstring = strtr($newstring, array('__PROTECTand__' => '&', '__PROTECTlt__' => '<', '__PROTECTgt__' => '>', '__PROTECTdquot__' => '"'));
+		if (strlen($stringtoencode) > 1000000) {// limit, to avoid memory problem
+			return htmlentities(substr($stringtoencode, 0, 1000000)) . '...';
```

Comment:
Using simple htmlentities will lead to security vulnerability.

Also the only thing we have after is return, so I don't see why returning a string instead of another (both are already in memory) generates a memory error

This means the $newstring is too large, so we must investigate what happen to have this memory error.